Docker dev environment

ramunasd opened the pull request:
Based on PHP5.4 as it is minimal supported version.

Fo the first time:
`docker-compose up -d`
`docker-compose exec php composer install`

to run tests:
`docker-compose exec php vendor/bin/phpunit`

lukebakken commented:
This seems like additional maintenance for this project. @hairyhum do you have an opinion?

ramunasd commented:
Well, I believe it is easiest way to start developement/testing for most developers, regardless operating system and package manager. Unless You still on old PHP 5.4 :)

ramunasd commented:
rebased and added https://github.com/Shopify/toxiproxy
With another dev dependency https://github.com/ihsw/toxiproxy-php-client we could simulate conditions like described in #624 #548 or even #437

lukebakken commented:
@ramunasd if you'd add examples of using `toxiproxy` then I think this could be useful for people who use docker and who don't. Thank you.

alcohol commented:
Oh I would love to see if this could help reproduce my problem (#548) more easily. @ramunasd got any pointers or documentation I could read, or even better, an example perhaps?

ramunasd commented:
@alcohol I can reproduce Your issue with non blocking streams and bigger data chunks. Will create separate PR for Toxiproxy based tests.

lukebakken commented on docker-compose.yaml:
```diff
@@ -0,0 +1,27 @@
+      - proxy
+
+  rabbitmq:
+    image: rabbitmq:3.5
```
What version of RabbitMQ does this image install? If it's RabbitMQ 3.5.X, that is very, very old. Can the latest one be used?

alcohol replied:
Yes that would install 3.5.x

https://hub.docker.com/_/rabbitmq

lukebakken replied:
That's too old, please use the latest available. Does Docker support something like `rabbitmq:latest` so that we don't have to keep modifying this file?

alcohol replied:
Yep it does. That should be fine for most scenarios I reckon.

ramunasd replied:
changed to latest

alcohol commented:
"Unfortunately" my issue is already resolved in the latest release I determined. So I am not sure you can reproduce it unless you test against an older version of the code :-)

lukebakken commented:
Thanks!

ramunasd commented:
Thank You @lukebakken this change makes testing/development more fluent 👍
